tekton/release.sh

```diff
@@ -0,0 +1,321 @@
+    [[ -z ${RELEASE_VERSION} ]] && {
+      echo "no target release"
+      exit 1
+    }
+  }
+  [[ ${RELEASE_VERSION} =~ v[0-9]+\.[0-9]*\.[0-9]+ ]] || {
+    echo "invalid version provided, need to match v\d+\.\d+\.\d+"
+    exit 1
+  }
+}
+
+whichCluster() {
+  read -r -e -p "Are you deploying on OpenShift (Y/n): " YESORNO
+  if [ "${YESORNO}" == "Y" ] || [ "${YESORNO}" == "y" ]; then
+    CLUSTER='openshift'
+    type -p oc >/dev/null || {
+      echo "'oc' need to be available"
+      exit 1
+    }
+  elif [ "${YESORNO}" == "N" ] || [ "${YESORNO}" == "n" ]; then
+    CLUSTER='kubernetes'
+  else
+    echo 'invalid input'
+    exit 1
+  fi
+}
+
+createGitTag() {
+  cd "${GOPATH}"/src/github.com/tektoncd/hub
+
+  [[ -n $(git status --porcelain 2>&1) ]] && {
+    echo "We have detected some changes in your repo"
+    echo "Stash them before executing this script"
+    exit 1
+  }
+
+  git checkout ${BRANCH}
+  git reset --hard ${UPSTREAM_REMOTE}/${BRANCH}
+
+  echo; echo 'Creating tag for new release:  '
+  read -r -e -p "Enter tag message: " TAG_MESSAGE
+  git tag -a "${RELEASE_VERSION}" -m "${TAG_MESSAGE}"
+  git push ${UPSTREAM_REMOTE} --tags
+}
+
+createHubSecretAndCM() {
+  kubectl create namespace ${HUB_NAMESPACE} 2>/dev/null || true
+
+  kubectl -n ${HUB_NAMESPACE} get secret db 2>/dev/null >/dev/null || {
+
+    echo; echo "Database Configurations:"
+    read -r -e -p "Enter DB Name: " DB_NAME
+    read -r -e -p "Enter DB Username: " DB_USERNAME
+    read -r -e -p "Enter DB Password: " DB_PASSWORD
+
+    kubectl -n ${HUB_NAMESPACE} create secret generic db \
+      --from-literal=POSTGRES_DB="${DB_NAME}" \
+      --from-literal=POSTGRES_USER="${DB_USERNAME}" \
+      --from-literal=POSTGRES_PASSWORD="${DB_PASSWORD}" \
+      --from-literal=POSTGRES_PORT="5432"
+
+    kubectl -n ${HUB_NAMESPACE} label secret db app=db
+    echo;
+  }
+
+  kubectl -n ${HUB_NAMESPACE} get secret api 2>/dev/null >/dev/null || {
+    echo "API Configurations:"
+    read -r -e -p "Enter GitHub OAuth Client ID: " GH_CLIENT_ID
+    read -r -e -p "Enter GitHub OAuth Client Secret: " GH_CLIENT_SECRET
+    read -r -e -p "Enter JWT Signing key: " JWT_SIGNING_KEY
+    read -r -e -p "Enter the Access JWT expire time: (eg. 1d) " ACCESS_JWT_EXPIRES_IN
+    read -r -e -p "Enter the Refresh JWT expire time: (eg. 1d) " REFRESH_JWT_EXPIRES_IN
+
+    kubectl -n ${HUB_NAMESPACE} create secret generic api \
+      --from-literal=GH_CLIENT_ID="${GH_CLIENT_ID}" \
+      --from-literal=GH_CLIENT_SECRET="${GH_CLIENT_SECRET}" \
+      --from-literal=JWT_SIGNING_KEY="${JWT_SIGNING_KEY}" \
+      --from-literal=ACCESS_JWT_EXPIRES_IN="${ACCESS_JWT_EXPIRES_IN}" \
+      --from-literal=REFRESH_JWT_EXPIRES_IN="${REFRESH_JWT_EXPIRES_IN}"
+
+    kubectl -n ${HUB_NAMESPACE} label secret api app=api
+
+    kubectl -n ${HUB_NAMESPACE} create cm ui \
+      --from-literal=GH_CLIENT_ID="${GH_CLIENT_ID}" \
+      --from-literal=API_URL="https://api.hub.tekton.dev"
```

Comment:
Should we make this also like ask from user?

Reply:
that field will have to be updated once API server is deployed ... will change cluster to cluster. 
this is for upstream only